### DISCUSSION TITLE
Historical tick / trade data?

### DISCUSSION THREAD

**Discussion by atulc** [Petersoj/IQFeed4j#10] [General]
Hi Peter,

First, thanks for the excellent library; I've been trying to use it to develop a screener and I had a question:

How would I retrieve historical TICK data? I know that in tape replay tools you can see bar movement at the tick level (sudden pulls/pushes in the tape in sub-second time as we see in, for example, trading view) but when I store the data it only stores is at the SECOND level, however on level1 feed I do get the tick data:

LEVEL 1:

![image](https://user-images.githubusercontent.com/2449316/194038663-7559799c-f12d-4a5f-af19-c297e14ad493.png)

`                try {
                    iqFeed4j.startLevel1Feed();
                    iqFeed4j.level1().waitForProtocolVersionValidation(5, TimeUnit.SECONDS);

                    iqFeed4j.level1().selectUpdateFieldNames(
                            SummaryUpdateField.ASK,
                            SummaryUpdateField.MOST_RECENT_TRADE_SIZE,
                            SummaryUpdateField.MOST_RECENT_TRADE_MARKET_CENTER,
                            SummaryUpdateField.MOST_RECENT_TRADE_DATE,
                            SummaryUpdateField.MOST_RECENT_TRADE_TIME);

                    iqFeed4j.level1().requestWatchTrades("SQQQ",
                            (fundamentalData) -> System.out.printf("Apple's most recent %.2f split was on %s.\n",
                                    fundamentalData.getSplitFactor1(), fundamentalData.getSplitFactor1Date()),
                            (summaryUpdate) -> System.out.printf("AAPL trade occurred with %d shares at $%.4f with " +
                                            "the market center code %d at %s %s with a latency of %s milliseconds.\n",

                                    summaryUpdate.getAsk(),
                                    summaryUpdate.getMostRecentTradeSize(),
                                    summaryUpdate.getMostRecentTrade(),
                                    summaryUpdate.getMostRecentTradeMarketCenter(),
                                    summaryUpdate.getMostRecentTradeDate(),
                                    summaryUpdate.getMostRecentTradeTime(),
                                    ChronoUnit.MILLIS.between(
                                            summaryUpdate.getMostRecentTradeTime(),
                                            LocalTime.now(ZoneId.of("America/New_York")))));
                } catch (IOException | TimeoutException exception) {
                    exception.printStackTrace();
                }`

HISTORICAL:

![image](https://user-images.githubusercontent.com/2449316/194039255-2e135846-ac73-45dd-abc3-f561c15b277f.png)

`            IQFeed4j iqFeed4j = new IQFeed4j("IQFeed", "localhost", 9100, 9200, 9400, 9300, 9100);
            iqFeed4j.startHistoricalFeed();

            if (1 == 1) {
                List<Interval> aaplIntervals = iqFeed4j.historical().requestIntervals(
                        SYMBOL,
                        2,
//                        LocalDateTime.of(2022, 5, 5, 4, 00),
//                        LocalDateTime.of(2022, 5, 5, 20, 0),
                        LocalDateTime.of(2022, 10, 05, 4, 0),
                        LocalDateTime.of(2022, 10, 05, 20, 0),
                        null,
                        LocalTime.of(4, 00),
                        LocalTime.of(20, 0),
                        DataDirection.OLDEST_TO_NEWEST,
                        IntervalType.TICKS);
                System.out.println(SYMBOL + "  minute intervals:");`

**Comment by Petersoj** [comment]
Hey @atulc,

Thanks for using the library! I'm glad you've found it useful.
For historical tick intervals, you'll want to use one of the [`HistoricalFeed.requestTicks()`](https://javadoc.io/doc/net.jacobpeterson/iqfeed4j/latest/net/jacobpeterson/iqfeed4j/feed/lookup/historical/HistoricalFeed.html#requestTicks(java.lang.String,int,java.lang.Integer,java.time.LocalTime,java.time.LocalTime,net.jacobpeterson.iqfeed4j.model.feed.lookup.historical.enums.DataDirection)) methods. See the `HistoricalFeed` Javadoc [here](https://javadoc.io/doc/net.jacobpeterson/iqfeed4j/latest/net/jacobpeterson/iqfeed4j/feed/lookup/historical/HistoricalFeed.html) for its full list of method signatures.

Thanks,

Jacob

**Reply by atulc** [reply]
Thank you, Peter!
